### PR TITLE
OnnxRuntime Eager: Implement log_softmax with ONNX Ops

### PR DESCRIPTION
**Pull request by msftlincoln**
**Description**: Provide implementation for `_log_softmax_out` using ONNX Ops `LogSoftmax` and `Transpose`, per definition [here](https://cs.github.com/pytorch/pytorch/blob/master/torch/onnx/symbolic_opset9.py#L1889).

- Move `CHECK_STATUS` macro to shared header (from generator) so it can be used by both generated and non-generated code.
- Test for log_softmax specifying different dimensions to verify both transpose and non-transpose cases.
- Implementation for `_log_softmax_out` using `SignatureOnly` prototype.

**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

**Comment by msftlincoln** [comment]
Confirmed the transpose vs not-transpose test cases using print statements (now removed).
Confirmed that not specifying a dim is invalid for CPU (i.e. no default case mismatch)

File: orttraining/orttraining/eager/ort_aten.h [hunk 1]
```diff
@@ -11,6 +11,8 @@
 #include "ort_log.h"
 #include "ort_tensor.h"
 
+#define CHECK_STATUS(status) if(!status.IsOK()) { std::stringstream err; err << "ORT return failure (line " << __LINE__ << "): " << status.ErrorMessage(); throw std::runtime_error(err.str()); }
```
Comment (WilBrady) [inline orttraining/orttraining/eager/ort_aten.h:14]: please fix the warning about log line length and spacing.

File: orttraining/orttraining/eager/test/ort_ops.py [hunk 2]
```diff
@@ -232,6 +232,17 @@ def test_softmax(self):
...
 
+    def test_log_softmax(self):
+        device = self.get_device()
+        cpu_tensor = torch.rand(3, 5)
+        ort_tensor = cpu_tensor.to(device)
+        cpu_result_a = torch.log_softmax(cpu_tensor, dim=1)
+        ort_result_a = torch.log_softmax(ort_tensor, dim=1)
+        assert torch.allclose(cpu_result_a, ort_result_a.cpu())
+        cpu_result_b = torch.log_softmax(cpu_tensor, dim=0)
+        ort_result_b = torch.log_softmax(ort_tensor, dim=0)
```
Comment (WilBrady) [inline orttraining/orttraining/eager/test/ort_ops.py:243]: Consider adding case with dim = -1 to exercise the dim < 0 code path or increase the test tensor dim to 3 (rand(3,5,5 for example) update the dim=0 case to dim =-1.

**Review by WilBrady — APPROVED** [review]

File: orttraining/orttraining/eager/test/ort_ops.py (hunk 2 above)
Comment (WilBrady) [reply orttraining/orttraining/eager/test/ort_ops.py:243]: also test the out param
